README.md

```diff
@@ -79,11 +79,57 @@ let line = data |> Js.Json.parseExn
 ```
 
 #### Encode and decode Base64
+
+To encode and decode Base64, you can bind to Javascript functions `btoa` and `atob`, respectively:
+
+```ml
+external  btoa : string -> string = "window.btoa" [@@bs.val]
+external  atob : string -> string = "window.atob" [@@bs.val]
+
+let _ =  "Hello World!" |> btoa |> atob |> Js.log
+```
+
 #### Generate random numbers
+
+Use [Random module](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Random.html) to generate random numbers
+
+```ml
+let () = 
+  Js.log @@ Random.int 5
+```
+
 #### Log a message to the console
+
+```ml
+Js.log "Hello BuckleScript!"
+```
+
 #### Use string interpolation
+
 #### Format a string using Printf
-#### Make and usa a Map
+
+Use [Printf module](http://caml.inria.fr/pub/docs/manual-ocaml/libref/Printf.html)
+
+```ml
+# Printf.printf ("Foo %d  %s") 2 "bar"
+```
+
+#### Make and use a Map
+
+To create a Map, use the Map.Make functor. It expects a module with the folowing signature:
+```ml
+module type OrderedType = sig type t val compare : t -> t -> int end
+```
+
+For instance, to create the map which associate 1 to "ocaml" and 2 to "bs":
+```ml
+let () = 
+  let module IntMap = 
+    Map.Make(struct type t = int let compare = compare end) in
+  let open IntMap in
+    add 1 "ocaml" (add 2 "bs" empty) |> 
+    iter (fun k v -> Js.log ("key: " ^ (string_of_int k) ^ " val: " ^ v))
```

Comment:
Could you separate the steps here a bit more? Use a variable instead of the pipe operator, add some whitespace in between them and a comment describing each step?